pennylane_lightning/src/util/TypeTraits.hpp

```diff
@@ -31,4 +31,28 @@ template <typename T> struct is_complex : std::false_type {};
 template <typename T> struct is_complex<std::complex<T>> : std::true_type {};
 
 template <typename T> constexpr bool is_complex_v = is_complex<T>::value;
+
+/**
+ * @brief Function return type
+ *
+ * Usage:
+ * .. code-block::cpp
+ *
+ *     std::pair<int, int> g(std::tuple<int, int, int>);
+ *     static_assert(std::is_same_v<FuncReturn<decltype(g)>::Type,
+ * std::pair<int,int>>); // return type of g is std::pair<int, int>
+ *
+ */
+
+template <class F> struct FuncReturn {
+    // When instantiated
+    static_assert(sizeof(F) == -1,
```

Comment:
Any reason about the `-1` here?

Reply:
This is just for raising errors when instantiated, i.e., only specialized classes can be used.

Reply:
Would a private/protected constructor be better here? Fine with the -1 for now, but wondering about long term intent declaration.